protocol_rfcs/catalog-owned.md

```diff
@@ -0,0 +1,489 @@
+  filesystem was dropped
+- commit 9 is an inline commit
+- neither of the two staged commits for version 10 have been ratified
+
+To read such tables, Delta clients must first contact the catalog to get the ratified commits. This
+informs the Delta client of commits [6, 9] as well as the latest ratified version, 9.
+
+If this information is insufficient to construct a complete snapshot of the table, Delta clients
+must LIST the `_delta_log` directory to get information about the published commits. For commits
+that are both returned by the catalog and already published, Delta clients can choose to read
+either. Delta clients must ignore any files with versions greater than the latest ratified commit
+version returned by the catalog.
+
+Combining these two sets of files and commits enables Delta clients to generate a snapshot at the
+latest version of the table.
+
+**NOTE**: This spec prescribes the _minimum_ required interactions between Delta clients and
+catalogs for commits. Catalogs may very well expose APIs and work with Delta clients to be
+informed of other non-commit [file types](#file-types), such as checkpoint, log
+compaction, and version checksum files. This would allow catalogs to return additional
+information to Delta clients during query and scan planning, potentially allowing Delta
+clients to avoid LISTing the filesystem altogether.
+
+## Commit Protocol
+
+To start, Delta Clients send the desired actions to be commited to the client-side component of the
+catalog.
+
+This component then has several options for proposing, ratifying, and publishing the commit,
+detailed below.
+
+- Option 1: Write the actions (likely client-side) to an [staged commit file](#staged-commit) in the
+  `_delta_log/_staged_commits` directory and then ratify the staged commit by atomically recording
+  (likely server-side) (in persistent storage of some kind) that the file corresponds to version
+  `v`.
+- Option 2: Treat this as an [inline commit](#inline-commit) (i.e. likely that the client-side
+  component sends the contents to the server-side component) and atomically record the content of
+  the commit as version `v` of the table.
+- Option 3: Propose, ratify, and publish all-in-one by atomically writing a [published commit file](#published-commit)
+  in the `_delta_log` directory. Note that this commit will be considered to have succeeded as soon
+  as the file becomes visible in the filesystem, regardless of when or whether the catalog is made
+  aware of the successful publish. The catalog does not need to track these files.
+
+A catalog must not ratify version `v` until it has ratified version `v - 1`, and it must ratify
+version `v` at most once.
+
+The catalog must track both flavors of ratified commits (staged or inline) and make them available
+to readers until they are [published](#publishing-commits).
+
+Delta clients are encouraged to establish an API contract where the catalog provides the latest
```

Comment:
Mention "for performance reasons"? Because it's really just a performance optimization.